chronicle/src/main/java/net/openhft/chronicle/tcp/SinkTcp.java

```diff
@@ -65,7 +84,8 @@ public boolean open(boolean blocking) throws IOException {
                 socketChannel.socket().setSendBufferSize(this.builder.sendBufferSize());
             }
 
-            super.setSocketChannel(socketChannel);
+            this.socketChannel = socketChannel;
+            this.tcpEventHandler = new TcpEventHandler(socketChannel, builder.tcpPipeline(sinkTcpHandler), sessionDetailsProvider, builder.sendBufferSize(), builder.receiveBufferSize());
```

Comment:
I think it is an error: the behavior should be consistent among sinks/remotes. 
Would it be possible to rework this patch to have the possibility to configure blocking/non-blocking consistently? It would be really appreciated